### PR TITLE
Make validity and genvalidity compatible with base-4.7

### PR DESCRIPTION
**Pull request by sjakobi**
As needed for https://github.com/commercialhaskell/path/issues/80.

### CODE REVIEW

File: validity/src/Data/Validity.hs [hunk 1]
```diff
@@ -1,8 +1,8 @@
+{-# LANGUAGE CPP #-}
 {-# LANGUAGE FlexibleInstances #-}
 {-# LANGUAGE TypeOperators #-}
 {-# LANGUAGE FlexibleContexts #-}
 {-# LANGUAGE DefaultSignatures #-}
-{-# OPTIONS_GHC -Wno-redundant-constraints #-}
```
Comment (NorfairKing) [inline validity/src/Data/Validity.hs]: Don't remove this please. We'll get a redundant constraints warning on `HasResolution`.

File: genvalidity/src/Data/GenValidity.hs [hunk 2]
```diff
@@ -46,7 +46,6 @@
 {-# LANGUAGE FlexibleContexts #-}
 {-# LANGUAGE TypeOperators #-}
 {-# LANGUAGE DefaultSignatures #-}
-{-# OPTIONS_GHC -Wno-redundant-constraints #-}
```
Comment (NorfairKing) [inline genvalidity/src/Data/GenValidity.hs]: Don't remove this please. See my other comment.

**Comment by NorfairKing** [comment]
Otherwise, this looks good. Will have another look tomorrow.

File: validity/src/Data/Validity.hs (hunk 1 above)
Comment (sjakobi) [inline validity/src/Data/Validity.hs]: I tried solving it like this but GHC didn't accept that:

```
#if __GLASGOW_HASKELL__ >= 800
{-# OPTIONS_GHC -Wno-redundant-constraints #-}
#endif
```

I have moved the options to the cabal file now.

**Comment by NorfairKing** [comment]
I'd prefer to turn them off on a per-module level (per constraint, really), but this is good enough, I guess.

**Pull request merged**